lib/diagnostic.js

```diff
@@ -36,7 +63,12 @@ const newArray = function newArray(length, defaultValue) {
 // The function you write should return
 // the object passed in as an argument or the object created
 const addMethod = function addMethod(propertyName, method, obj) {
-
+if(obj === "undefined"){
+  obj = {};
+}else{
+  obj.propertyName.method = method;
+}
+return obj;
```

Comment:
This is getting declaration exceptions, because they are undefined.  Instead of checking if obj is undefined we can write `if (!obj)`.  use `obj[propertyName] = method;` instead of `obj.propertyName.method = method;`, and you don't need to place it in an `else { }`.